### PR TITLE
[mmp] Add support for linking only the platform (Xamarin.Mac.dll) assembly on the full profile

### PR DESCRIPTION
**Pull request by spouliot**
Replace https://github.com/xamarin/xamarin-macios/pull/1973 expect that
the test parts are still needed.

### CODE REVIEW

**Comment by spouliot** [comment]
@VincentDondain ^ this will require an update to the Mac addin UI (for `d15-3`)

**Comment by spouliot** [comment]
I'll create a card for QA to test this

**Comment by chamons** [comment]
Building at least a few large mac samples (SourceWriter, etc) in XM 4.5 (Full) and with --linkplatform  in mmp args would be grand.

**Comment by spouliot** [comment]
https://trello.com/c/fjZN5XMf/9-xm-link-platform-on-full-profile

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -274,6 +274,7 @@ static void Main2 (string [] args)
 						}
 					}
 				},
+				{ "linkplatform", "Link only the Xamarin.Mac.dll platform assembly", v => App.LinkMode = LinkMode.Platform },
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: What about XamMac.dll? Do we want to support Classic? We might want to disallow it, just to minimize the test matrix.

Comment (chamons) [inline tools/mmp/driver.cs]: Not a bad point. Let me add an error...

Comment (chamons) [inline tools/mmp/driver.cs]: @rolfbjarne done in https://github.com/xamarin/xamarin-macios/pull/1990/commits/acdbe743d7bf7ef5ce1e4892c3d725d875b61b6a

File: tools/mmp/error.cs [hunk 2]
```diff
@@ -73,6 +73,7 @@ namespace Xamarin.Bundler {
 	//				Warning	MM2016  Invalid TlsProvider `{0}` option. The only valid value `{1}` will be used.
 	//					MM2017  Could not process XML description: {0}
 	//					MM202x	Binding Optimizer failed processing `...`.
+	//					MM2100  Xamarin.Mac Classic API does not support Platofmr Linking.
```
Comment (abock) [inline tools/mmp/error.cs]: Typo ("Platform")

Comment (chamons) [inline tools/mmp/error.cs]: Fixed in merge with master. Thanks!

**Comment by chamons** [comment]
Master merged but not built locally. Waiting for bots to chew on it.

**Comment by spouliot** [comment]
@chamons merge issue ? all tests fails to build due to 
> 	/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/common/mac/ProjectTestHelpers.cs(243,97): error CS0103: The name 'GenerateOuputCommand' does not exist in the current context

**Comment by chamons** [comment]
Yep. On it.

**Pull request merged**